src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -104,512 +158,931 @@ enum Spells
+                            break;
+                        default:
+                            break;
+                    }
+                }
+
+                if (!UpdateVictim())
+                    return;
+
+                DoMeleeAttackIfReady();
             }
-        }
 
-        void MovementInform(uint32 MovementType, uint32 /*Data*/) override
+        private:
+            bool _faceroller;
+        };
+
+        CreatureAI* GetAI(Creature* creature) const override
         {
-            if (MovementType != POINT_MOTION_TYPE)
-                return;
+            return GetTrialOfChampionAI<boss_eadricAI>(creature);
+        }
+};
 
-            instance->SetData(BOSS_ARGENT_CHALLENGE_E, DONE);
+enum PaletressData
+{
+    DATA_MEMORY_ENTRY = 1,
 
-            me->DisappearAndDie();
-        }
+    ACTION_MEMORY_DEATH
+};
+
+class boss_paletress : public CreatureScript
+{
+    public:
+        boss_paletress() : CreatureScript("boss_paletress") { }
 
-        void UpdateAI(uint32 uiDiff) override
+        struct boss_paletressAI : public argent_challenge_baseAI
         {
-            if (bDone && uiResetTimer <= uiDiff)
+            boss_paletressAI(Creature* creature) : argent_challenge_baseAI(creature) { }
+
+            void Initialize() override
             {
-                me->GetMotionMaster()->MovePoint(0, 746.87f, 665.87f, 411.75f);
-                bDone = false;
-            } else uiResetTimer -= uiDiff;
+                argent_challenge_baseAI::Initialize();
+                _memorySummoned = false;
+            }
 
-            if (!UpdateVictim())
-                return;
+            void OnReset() override
+            {
+                me->RemoveAura(SPELL_SHIELD);
+                Initialize();
+                argent_challenge_baseAI::OnReset();
+            }
 
-            if (uiHammerJusticeTimer <= uiDiff)
+            void DoAction(int32 action) override
             {
+                if (action != ACTION_MEMORY_DEATH)
+                    return;
+
                 me->InterruptNonMeleeSpells(true);
+                Talk(SAY_MEMORY_DEATH);
+                me->GetMotionMaster()->Clear();
+                me->RemoveAura(SPELL_SHIELD);
+            }
+
+            uint32 GetData(uint32 type) const override
+            {
+                if (type != DATA_MEMORY_ENTRY || _memoryGuid.IsEmpty())
+                    return 0;
 
-                if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 250, true))
+                return _memoryGuid.GetEntry();
+            }
+
+            void DamageTaken(Unit* /*done_by*/, uint32 &damage) override
+            {
+                if (instance->GetBossState(DATA_ARGENT_CHALLENGE) == DONE)
                 {
-                    if (target->IsAlive())
+                    damage = 0;
+                    return;
+                }
+
+                if (!_memorySummoned && !HealthAbovePct(25))
```

Comment:
`HealthAbovePctDamaged`, maybe?

Reply:
also, consider capping damage to keep her at just under 25% HP here - otherwise you can still overkill to skip this phase

Reply:
@Treeston  HealthAbovePctDamaged? I cant find this function anywhere

Reply:
`HealthBelowPctDamaged`, my bad.

Do Something like
```c++
if (!_memorySummoned && me->HealthBelowPctDamaged(25, damage))
{
    damage = me->GetHealth() - me->CountPctFromMaxHealth(25);
    /* summon etc */
```